legate/info.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2021-2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# mypy: ignore-errors
+import json
+
+from IPython.core.magic import Magics, line_magic, magics_class
+
+cmd_dict = {
+    "cpus": "Number of CPUs to use per rank",
+    "gpus": "Number of GPUs to use per rank",
+    "openmp": "Number of OpenMP groups to use per rank",
+    "ompthreads": "Number of threads per OpenMP group",
+    "utility": "Number of Utility processors per rank",
+    "sysmem": "Amount of DRAM memory per rank (in MBs)",
+    "numamem": "Amount of DRAM memory per NUMA domain per rank (in MBs)",
+    "fbmem": "Amount of framebuffer memory per GPU (in MBs)",
+    "zcmem": "Amount of zero-copy memory per rank (in MBs)",
+    "regmem": "Amount of registered CPU-side pinned memory per rank (in MBs)",
+    "nodes": "Number of nodes to use",
+}
+
+
+class LegateInfo(object):
+    def __init__(self, filename: str) -> None:
+        self.config_dict = dict()
+        with open(filename) as json_file:
+            json_dict = json.load(json_file)
+            for key in cmd_dict.keys():
+                if key not in json_dict:
+                    assert 0, "Key: " + key + " does not exist."
+                else:
+                    self.config_dict[key] = json_dict[key]["value"]
+
+    def __repr__(self) -> str:
+        out_str = ""
+        for key, value in self.config_dict.items():
+            out_str += cmd_dict[key] + ": " + str(value) + "\n"
+        out_str = out_str[:-1]
+        return out_str
+
+
+_legate_info = LegateInfo("legate_jupyter.json")
```

Comment:
How is the json file found? Would the user always have to run from inside the main legate.core source directory?

Reply:
Now, I have pushed another change. 
We will move the `install_jupyter_kernel.py` into `LEGATE_DIR/bin`, then users can run
```
LEGATE_DIR/bin/install_jupyter_kernel --json=legate_jupyter.json
```
`--json=legate_jupyter.json` is optional if the file is already exists in the current directory. 
After installation, we will copy the json file into the ipython kernel installation directory, which is `/home/wwu/.local/share/jupyter/kernels/legate_kernel_nocr`, the the `info.py` can load the json file from there.

Reply:
>  if the file is already exists in the current directory.


It seems like the default JSON file should be included in the package distribution as package data, then it can be located/loaded related to this module's `__file__`, regardless of what cwd is (which could be anything at all)

Alternatively, it's possible there is something I am not grasping about the intended/desired UX

Reply:
Actually, isn't the point of `%legate_info` to print information about an *installed* legate kernel? Why do we need to print out anything at all if no JSON  file is found in the ipython kernel directory? Couldn't we just error in that case?

Reply:
I have refactor the info.py. Now the structure of the directory is:
```
legate/jupyter:
- __init__.py
- _magic_cmd.py
```
In order to use the magic cmd, users will run
```
%load_ext legate.jupyter
%legate_info
```

> Actually, isn't the point of %legate_info to print information about an installed legate kernel? Why do we need to print out anything at all if no JSON file is found in the ipython kernel directory? Couldn't we just error in that case?

This one has also been fixed.